File: backend/app/schemas/platform.py
from datetime import datetime, date
from typing import List, Optional
from uuid import UUID
from pydantic import BaseModel, ConfigDict, Field

class NotificationLogResponse(BaseModel):
    id: UUID
    user_id: UUID
    title: str
    message: str
    channel: str
    priority: str
    delivery_status: str
    read_status: bool
    created_at: datetime
    model_config = ConfigDict(from_attributes=True)

class WorkflowTriggerCreate(BaseModel):
    trigger_type: str
    event_name: str
    parameters: Optional[dict] = None

class WorkflowConditionCreate(BaseModel):
    operator: str = "AND"
    expression: dict

class WorkflowActionCreate(BaseModel):
    action_type: str
    parameters: dict
    sequence_order: int = 1

class WorkflowDefinitionCreate(BaseModel):
    name: str = Field(..., max_length=255)
    code: str = Field(..., max_length=100)
    description: Optional[str] = None
    triggers: List[WorkflowTriggerCreate]
    conditions: List[WorkflowConditionCreate]
    actions: List[WorkflowActionCreate]

class WorkflowDefinitionResponse(BaseModel):
    id: UUID
    name: str
    code: str
    version: int
    description: Optional[str]
    status: str
    created_at: datetime
    model_config = ConfigDict(from_attributes=True)

class AuditEventResponse(BaseModel):
    id: UUID
    entity_name: str
    entity_id: UUID
    operation: str
    old_value: Optional[str]
    new_value: Optional[str]
    source: str
    user_id: Optional[UUID]
    request_id: Optional[str]
    ip_address: Optional[str]
    browser: Optional[str]
    created_at: datetime
    model_config = ConfigDict(from_attributes=True)

class FileMetadataResponse(BaseModel):
    id: UUID
    name: str
    folder_id: Optional[UUID]
    mime_type: str
    file_size_bytes: int
    storage_path: str
    sha256_checksum: str
    created_at: datetime
    model_config = ConfigDict(from_attributes=True)

class ScheduledJobResponse(BaseModel):
    id: UUID
    name: str
    code: str
    cron_expression: str
    status: str
    last_run_at: Optional[datetime]
    next_run_at: Optional[datetime]
    model_config = ConfigDict(from_attributes=True)

class SystemSettingResponse(BaseModel):
    id: UUID
    key: str
    value: str
    category: str
    description: Optional[str]
    model_config = ConfigDict(from_attributes=True)

class FeatureFlagResponse(BaseModel):
    id: UUID
    name: str
    enabled: bool
    description: Optional[str]
    model_config = ConfigDict(from_attributes=True)

class SearchIndexResponse(BaseModel):
    id: UUID
    entity_type: str
    entity_id: UUID
    title: str
    description: Optional[str]
    metadata_json: Optional[dict]
    model_config = ConfigDict(from_attributes=True)

class SystemEventResponse(BaseModel):
    id: UUID
    event_name: str
    correlation_id: UUID
    parent_event_id: Optional[UUID]
    version: int
    payload: dict
    status: str
    duration_ms: int
    created_at: datetime
    model_config = ConfigDict(from_attributes=True)

class HealthMetricResponse(BaseModel):
    id: UUID
    api_latency_ms: float
    db_latency_ms: float
    redis_connected: bool
    disk_usage_percent: float
    memory_usage_percent: float
    scheduler_queue_depth: int
    email_queue_depth: int
    workflow_queue_depth: int
    created_at: datetime
    model_config = ConfigDict(from_attributes=True)

File: frontend/src/types/platform.ts
export interface NotificationLog {
  id: string;
  user_id: string;
  title: string;
  message: string;
  channel: 'IN_APP' | 'EMAIL' | 'SMS' | 'WHATSAPP' | 'PUSH';
  priority: 'LOW' | 'NORMAL' | 'HIGH' | 'URGENT';
  delivery_status: 'PENDING' | 'SENT' | 'FAILED';
  read_status: boolean;
  created_at: string;
}

export interface WorkflowTrigger {
  id: string;
  trigger_type: string;
  event_name: string;
  parameters?: Record<string, any>;
}

export interface WorkflowCondition {
  id: string;
  operator: string;
  expression: Record<string, any>;
}

export interface WorkflowAction {
  id: string;
  action_type: string;
  parameters: Record<string, any>;
  sequence_order: number;
}

export interface WorkflowDefinition {
  id: string;
  name: string;
  code: string;
  version: number;
  description?: string;
  status: string;
  created_at: string;
}

export interface AuditEvent {
  id: string;
  entity_name: string;
  entity_id: string;
  operation: string;
  old_value?: string;
  new_value?: string;
  source: 'API' | 'SCHEDULER' | 'WORKFLOW' | 'AI' | 'MANUAL';
  user_id?: string;
  request_id?: string;
  ip_address?: string;
  browser?: string;
  created_at: string;
}

export interface FileMetadata {
  id: string;
  name: string;
  folder_id?: string;
  mime_type: string;
  file_size_bytes: number;
  storage_path: string;
  sha256_checksum: string;
  created_at: string;
}

export interface ScheduledJob {
  id: string;
  name: string;
  code: string;
  cron_expression: string;
  status: 'ACTIVE' | 'INACTIVE';
  last_run_at?: string;
  next_run_at?: string;
}

export interface SystemSetting {
  id: string;
  key: string;
  value: string;
  category: string;
  description?: string;
}

export interface FeatureFlag {
  id: string;
  name: string;
  enabled: boolean;
  description?: string;
}

export interface HealthMetric {
  id: string;
  api_latency_ms: number;
  db_latency_ms: number;
  redis_connected: boolean;
  disk_usage_percent: number;
  memory_usage_percent: number;
  scheduler_queue_depth: number;
  email_queue_depth: number;
  workflow_queue_depth: number;
  created_at: string;
}

export interface SearchIndex {
  id: string;
  entity_type: string;
  entity_id: string;
  title: string;
  description?: string;
  metadata_json?: Record<string, any>;
}
